tests/plotting/test_plot_manager.py

```diff
@@ -406,3 +407,57 @@ async def test_plot_info_caching(test_environment):
     await run_refresh_test(plot_manager)
     assert len(plot_manager.plots) == len(plot_manager.plots)
     plot_manager.stop_refreshing()
+
+
+@pytest.mark.parametrize(
+    ["event_to_raise"],
+    [
+        pytest.param(PlotRefreshEvents.started, id="started"),
+        pytest.param(PlotRefreshEvents.batch_processed, id="batch_processed"),
+        pytest.param(PlotRefreshEvents.done, id="done"),
+    ],
+)
+@pytest.mark.asyncio
+async def test_callback_event_raises(test_environment, event_to_raise: PlotRefreshEvents):
+    last_even_fired: Optional[PlotRefreshEvents] = None
+
+    def raising_callback(event: PlotRefreshEvents, _: PlotRefreshResult):
+        nonlocal last_even_fired
+        last_even_fired = event
+        if event == event_to_raise:
+            raise Exception(f"run_raise_in_callback {event_to_raise}")
+
+    env: TestEnvironment = test_environment
+    # Load dir_1
+    add_plot_directory(env.root_path, str(env.dir_1.path))
+    expected_result.loaded = len(env.dir_1)
+    expected_result.removed = 0
+    expected_result.processed = len(env.dir_1)
+    expected_result.remaining = 0
+    await run_refresh_test(env.plot_manager)
+    # Load dir_2
+    add_plot_directory(env.root_path, str(env.dir_2.path))
+    expected_result.loaded = len(env.dir_2)
+    expected_result.removed = 0
+    expected_result.processed = len(env.dir_1) + len(env.dir_2)
+    expected_result.remaining = 0
+    await run_refresh_test(env.plot_manager)
+    # Now raise the exception in the callback
+    default_callback = env.plot_manager._refresh_callback
+    env.plot_manager.set_refresh_callback(raising_callback)
+    await run_refresh_test(env.plot_manager, expect_match=False)
+    # And make sure the follow-up evens aren't fired
+    assert last_even_fired is not None and last_even_fired.value == event_to_raise.value
+    # The exception should trigger `PlotManager.reset()` and clear the plots
+    assert len(env.plot_manager.plots) == 0
+    assert len(env.plot_manager.plot_filename_paths) == 0
+    assert len(env.plot_manager.failed_to_open_filenames) == 0
+    assert len(env.plot_manager.no_key_filenames) == 0
+    # The next run without the valid callback should lead to re-loading of all plot
+    env.plot_manager.set_refresh_callback(default_callback)
+    expected_result.loaded = len(env.dir_1) + len(env.dir_2)
+    expected_result.removed = 0
+    expected_result.processed = len(env.dir_1) + len(env.dir_2)
+    expected_result.remaining = 0
+    await run_refresh_test(env.plot_manager)
+    env.plot_manager.stop_refreshing()
```

Comment:
Sounds like it makes sense but a bit unrelated to this PR.

Reply:
A `try`/`finally` then?  Unless it doesn't really matter if it runs in which case we can just delete the call.  But, I think if this doesn't run it will leave the thread laying around?

Another task for another PR, it needs to turn async and poll for the thread ending.  As is, it's just an indefinitely blocking call.

Reply:
> A `try`/`finally` then?

I would rather just leave it as it is for now unless it brings some real risk/issue with it other than the failed test not stopping properly. The other tests in that file have the same issue which anyway needs to be addressed separately then.